test/block.js

```diff
@@ -1,221 +1,88 @@
-    u.readOnceFromDB(alice),
+    pify(alice.publish)({ type: 'post', text: 'hello' }),
+    pify(bob.publish)({ type: 'post', text: 'hello' })
   ])
 
-  // should be the alice's follow(bob) message.
-  t.equal(msgAtBob.value.author, alice.id, 'bob received message from alice')
-  t.equal(msgAtBob.value.content.contact, bob.id, 'received message is about bob')
-
-  // should be the bob's follow(alice) message.
-  t.equal(msgAtAlice.value.author, bob.id, 'alice received message from bob')
-  t.equal(msgAtAlice.value.content.contact, alice.id, 'received message is about alice')
-
-  // disconnect bob from alice
-  await pify(rpcBobToAlice.close)(true)
+  // Self replicate
+  alice.ebt.request(alice.id, true)
+  bob.ebt.request(bob.id, true)
 
-  // alice blocks bob
-  await pify(alice.publish)(u.block(bob.id))
-  t.pass('alice published a block on bob')
+  alice.ebt.request(bob.id, false)
+  alice.ebt.block(alice.id, bob.id, true)
+  t.pass('alice does not want bob\'s data')
 
-  // get alice's follow graph
-  const g = await pify(alice.friends.get)(null)
-  t.equal(g[alice.id][bob.id], false, 'alice indicates she blocks bob')
+  bob.ebt.request(alice.id, true)
+  bob.ebt.block(bob.id, alice.id, false)
+  t.pass('bob wants alice\'s data')
 
-  // since bob is blocked, he should not be able to connect to alice
   try {
-    await pify(bob.connect)(alice.getAddress())
-    t.fail('bob.connect succeeded but it should have failed')
+    await Promise.all([
+      pify(bob.connect)(alice.getAddress()),
+      u.readOnceFromDB(bob, REPLICATION_TIMEOUT),
+      u.readOnceFromDB(alice, REPLICATION_TIMEOUT)
+    ])
+    t.fail('replication should not succeed')
   } catch (err) {
-    t.match(
-      err.message,
-      /server hung up/,
-      'bob is blocked, should fail to connect to alice'
-    )
+    t.match(err.message, /readOnceFromDB timed out/, 'db reading timed out')
   }
 
-  // but carol is allowed to connect, because she is not blocked
-  const rpcCarolToAlice = await pify(carol.connect)(alice.getAddress())
-  await sleep(REPLICATION_TIMEOUT)
-
-  await pify(rpcCarolToAlice.close)(true)
-
-  // carol has replicated with alice
-  const clockCarol = await pify(carol.getVectorClock)()
-  t.equal(clockCarol[alice.id], 2, 'carol replicated everything from alice')
-
-  // alice does not follow carol, so didnt get data
   const clockAlice = await pify(alice.getVectorClock)()
-  t.notOk(clockAlice[carol.id], 'alice did not replicate alice')
-
-  t.end()
-})
-
-tape('carol does not replicate alice\'s data with bob', async (t) => {
-  t.plan(3)
-  // first, carol should have already replicated with alice.
-  // emits this event when did not allow bob to get this data.
-  const rpcBobToCarol = await pify(bob.connect)(carol.getAddress())
-  await sleep(REPLICATION_TIMEOUT)
-
-  await pify(rpcBobToCarol.close)(true)
+  t.notOk(clockAlice[bob.id], 'alice did not replicate bob')
 
+  // TODO: shouldn't bob be forbidden to replicate alice???
```

Comment:
this is testing bob in the clock, so this is fine :)

Reply:
Done